buildbot-host/buildmaster/master.cfg

```diff
@@ -494,12 +497,25 @@ for combo in compile_config_opt_combos:
     factories.update({factory_name: (factory, "unix", "default", "openvpn")})
     del factory
 
+# OpenVPN 2 --dev null server tests on Unix-style operating systems
+if openvpn_run_tserver_null_tests:
+    for combo in build_and_test_config_opt_combos:
+        factory = util.BuildFactory()
+        factory = openvpnAddCommonUnixStepsToBuildFactory(factory, combo, ccache)
+        factory = openvpnAddUnixCompileStepsToBuildFactory(factory, combo, ccache)
+        factory = openvpnAddDeviceSetupStepsToBuildFactory(factory, combo, ccache)
+        factory = openvpnAddTServerNullStepsToBuildFactory(factory, combo, ccache)
+        factory_name = getFactoryName("%s-tserver_null" % (combo))
+        factories.update({factory_name: (factory, 'unix', 'default', 'openvpn')})
```

Comment:
I'm not convinced that we need a new factory for this. I would just integrate that into the tclient factory.

Reply:
I remove the separate build type and tested and it seems to work.

The end result feels unhygienic, though, as t_server_null and t_client are now entangled a bit too much for my taste. If you're not opposed, I can refactor the order of "build and test" events along these lines:

1. Prepare for t_server_nul
2. Prepare for t_client
3. Run "make check" (runs t_client and t_server_null tests)
4. Cache results from t_client

Also, I just remembered the main reason I split the t_server_null tests into a separate build type: I have not been able (yet) to test t_server_null on all the *BSD variants let alone things like Solaris derivatives. So, when we merge this things might explode.

Reply:
Fine with refactoring the steps further, sounds good to me.

You could keep the global flag for enabling/disabling the new tests (you can make adding the step to the factory conditional on that), so that we can disable them easily when they explode and sort out the fallout before reenabling them.

Reply:
Ok!